src/super_gradients/training/datasets/samplers/class_balanced_sampler.py

```diff
@@ -0,0 +1,151 @@
+        oversample_threshold: Optional[float] = None,
+        oversample_aggressiveness: float = 0.5,
+    ) -> List[float]:
+        """
+        Oversampling scarce classes from detection dataset, following sampling strategy described in https://arxiv.org/pdf/1908.03195.pdf.
+
+        :param class_information_provider:      An object (probably a dataset) that provides the class information.
+        :param oversample_threshold:            A frequency threshold (fraction, 0-1). Classes that are *less frequent* than this threshold will be oversampled.
+                                                The default value is None. If None, the median of the class frequencies will be used.
+        :param oversample_aggressiveness:       How aggressive the oversampling is. The higher the value, the more aggressive the oversampling is.
+                                                The default value is 0.5, and corresponds to the implementation in the paper.
+                                                A value of 0.0 corresponds to no oversampling.
+
+
+        The repeat factor is computed as followed:
+        1. For each class c, compute the fraction # of images that contain it (its frequency): :math:`f(c)`
+        2. For each class c, compute the category-level repeat factor: :math:`r(c) = max(1, aggressiveness(threshold/f(c)))`
+        3. For each image I, compute the image-level repeat factor: :math:`r(I) = max_{c in I} r(c)`
+
+        Returns a list of repeat factors (length = dataset_length). How to read: result[i] is a float, indicates the repeat factor of image i.
+        """
+
+        class_information = class_information_provider.get_dataset_classes_information()
+
+        # 1. For each category c, compute the fraction # of images that contain it: f(c)
+        class_frequencies = np.sum(class_information, axis=0)
+        class_frequencies = class_frequencies / len(class_information)
+
+        # 2. For each class c, compute the class-level repeat factor: r(c) = max(1, sqrt(t/f(c)))
+        category_repeat = {
+            cat_id: cat_repeat
+            for cat_id, cat_repeat in enumerate(_default_oversample_heuristic(class_frequencies, oversample_threshold, oversample_aggressiveness))
+        }  # dict for ease of debugging
+
+        # 3. For each image I, compute the image-level repeat factor: r(I) = max_{c in I} r(c)
+        repeat_factors = list()
+        categories = np.arange(class_information.shape[1])
+        for sample_cat_freq in class_information:
+            cat_ids = categories[sample_cat_freq != 0]
+            if len(cat_ids) == 0:  # in case image doesn't have annotations, we will not over-sample nor ignore it
+                repeat_factors.append(1.0)
+            else:
+                repeat_factors.append(max({category_repeat[cat_id] for cat_id in cat_ids}))
+
+        return repeat_factors  # len = dataset_length
+
+    @staticmethod
+    def precompute_sample_repeat_factors(
```

Comment:
What is the point of this method? I don't quite get the motivation to save precompute factors in some file.